File: frontend/web/src/api/products/createProduct.ts
import axiosInstance from "../../utils/AxiosInstance.ts"
import type { Product } from "../../components/shared/interface.ts";

const createProduct = async ({ product } : { product: Product }) => {

    const token = localStorage.getItem('token');

    const { data } = await axiosInstance.post('/api/products', product, {
        headers: { authorization: `Bearer ${token}` }
    });

    return data.product;
}

export default createProduct;

File: frontend/desktop/python-client/PyQt5App/api/products/createProduct.py
import requests
from PyQt5App.utils.Address import address
from PyQt5App.utils.sessionManager import SessionManager

class CreateProduct:
    def __init__(self, product):
        self.session = SessionManager()
        self.address = address()
        self.product = product
        self.token = self.session.get_token()

    def createProduct(self):
        requests.post(
            f"{self.address}/api/products",
            json={
                "product": self.product
            },
            headers={
                "Authorization": f"Bearer {self.token}"
            }
        )